SSdecay - Making the station dirty(and creepy)

Gandalf2k15 opened the pull request:
## About The Pull Request

This system is incredibly intensive and should only ever be run once, which is why it is SS_NO_FIRE.

Basically what this does is it iterates through every part of the station and adjusts some area unique degradation, as well as some creepy aspects.

For instance, medical would start off with blood and perhaps vomit. 

This also adds flickering lights, to make a light flicker at round start, change the roundstart_flicker variable on the light.

## Changelog
:cl:
add: A subsystem has been added that allows the station to start dirty.
/:cl:

tf-4 commented on modular_skyrat/modules/decay_subsystem/code/decaySS.dm:
```diff
@@ -0,0 +1,83 @@
+	do_medical()
+
+/datum/controller/subsystem/decay/proc/do_common()
+	for(var/turf/open/iteraing_floor in possible_turfs)
+		if(prob(FLOOR_DIRT_PERCENT_CHANCE))
+			new /obj/effect/decal/cleanable/dirt(iteraing_floor)
```
iteraing_floor -> iterating_floor

tf-4 commented on modular_skyrat/modules/decay_subsystem/code/decaySS.dm:
```diff
@@ -0,0 +1,83 @@
+		if(prob(FLOOR_DIRT_PERCENT_CHANCE))
+			new /obj/effect/decal/cleanable/dirt(iteraing_floor)
+
+	for(var/turf/closed/iteraing_wall in possible_turfs)
+		if(prob(WALL_RUST_PERCENT_CHANCE))
+			var/mutable_appearance/rust = mutable_appearance(iteraing_wall.icon, "rust")
+			iteraing_wall.add_overlay(rust)
```
iteraing_wall -> iterating_wall

Gandalf2k15 replied:
OH DAMN SON ITS NOT AN ITER

RatFromTheJungle commented:
pr upstreem first 🛩️

hugmeorelse commented:
:)

OrionTheFox commented on modular_skyrat/modules/decay_subsystem/code/decaySS.dm:
```diff
@@ -0,0 +1,86 @@
+			if(prob(LIGHT_FLICKER_PERCENT_CHANCE * severity_modifier))
+				iterating_light.start_flickering()
+
+/datum/controller/subsystem/decay/proc/do_engineering()
```
messes spawn in the atmos air tanks, which are impossible to clean. any way to blacklist them from spawning anything, by chance?

Gandalf2k15 replied:
Shouldn't matter since it doesn't create atmos stuff, but yeah it can be blacklisted

OrionTheFox commented on modular_skyrat/modules/decay_subsystem/code/decaySS.dm:
```diff
@@ -0,0 +1,86 @@
+*/
+
+
+#define WALL_RUST_PERCENT_CHANCE 15
```
if you make rust something that happens a lot you might want to consider a way to remove the rust easily too, tho that's out of the PRs scope, just a note

Gandalf2k15 replied:
Easiest way is tooling them

OrionTheFox replied:
doesnt work well with reinforced walls
i just wanna clean 😢

Gandalf2k15 replied:
Eee there needs to be a rust component then

CandleJaxx commented:
the test merge on this was neat as heck but i also think that having the station ALWAYS start dirty is a biiit much-- maybe keep the standard hallways clean, and some spots of dirt and grime in departments / maint only

Gandalf2k15 commented:
> the test merge on this was neat as heck but i also think that having the station ALWAYS start dirty is a biiit much-- maybe keep the standard hallways clean, and some spots of dirt and grime in departments / maint only

Yeah I realised there was an issue with the code, it didn't have the ability not to happen.

Gandalf2k15 commented:
# Still undergoing massive finetuning.

OrionTheFox commented on code/controllers/subsystem/explosions.dm:
```diff
@@ -373,8 +373,10 @@ SUBSYSTEM_DEF(explosions)
-
-
+		//SKYRAT EDIT ADDITION
+		for(var/obj/machinery/light/iterating_light in T)
```
I assume it isn't intentional that this toggles low-power/dying lights to full and then off?
https://gyazo.com/45edfff25390ca159fca4cc8f08df05f

OrionTheFox replied:
oh, i actually think this is maybe to be flickering on/off but is flickering on/lowpower..

Gandalf2k15 replied:
It was fixed, server not updated

Arctaisia commented:
I like this PR quite a bit! Though ofcourse finetuning is being done (even better), it certainly does feel a bit overly rusty, though with it being fixed to be able to not happen, I can see much less rusted walls dotted about, and it should turn out to be good in the end regardless, thanks nerd! Some sort of rust removal tool for janitors/engies would be nice, but that's for later, *weh

BurgerLUA commented:
The intent of the PR is fine. Station dirt is a nice touch but it gives things for people to do, however there are some issues I think with the rest of the PR that I don't think are good for the server.

I have three issues with this PR:

- The amount of hostile mobs that spawn in maint should be limited to 1 giant spiders or bat per maint segment. I did a PR like this for Bastion of Hestia that did exactly this and it was neatly balanced and didn't turn into Security/Shaft Miners hunting mobs every round. You need to make it so that the amount of spiders and such are decently common, but not too common to the point where security cares.
- Lighting ballast malfunctions are not a good idea in a 2D game and especially not with this lighting subsystem. I thought it was a /tg/ upstream thing at once and declared that /tg/ was probably trolling everyone with the addition, but apparently it's a Skyrat thing. The issue here is that it's very difficult to actually locate lighting ballast malfunctions as not everyone has a surround sound system. It's also very annoying to hear that constant FLANG and you don't know where it's from. From a performance perspective, it's kind of insane. as well as you're forcing the lighting engine to update constantly. Sometimes on maps like delta, the lighting doesn't even update because of the flickering because it's under a lot of stress. There was a reason why flashlights, a usually constantly moving lightsource, were converted into lighting sprites instead of using the lighting subsystem.
- The mob spawns don't have numerical assignments, like Spider (120), and thus can cause issues when it comes to reporting players who break rules.

Gandalf2k15 commented:
> The intent of the PR is fine. Station dirt is a nice touch but it gives things for people to do, however there are some issues I think with the rest of the PR that I don't think are good for the server.
> 
> I have three issues with this PR:
> 
> * The amount of hostile mobs that spawn in maint should be limited to 1 giant spiders or bat per maint segment. I did a PR like this for Bastion of Hestia that did exactly this and it was neatly balanced and didn't turn into Security/Shaft Miners hunting mobs every round. You need to make it so that the amount of spiders and such are decently common, but not too common to the point where security cares.
> * Lighting ballast malfunctions are not a good idea in a 2D game and especially not with this lighting subsystem. I thought it was a /tg/ upstream thing at once and declared that /tg/ was probably trolling everyone with the addition, but apparently it's a Skyrat thing. The issue here is that it's very difficult to actually locate lighting ballast malfunctions as not everyone has a surround sound system. It's also very annoying to hear that constant FLANG and you don't know where it's from. From a performance perspective, it's kind of insane. as well as you're forcing the lighting engine to update constantly. Sometimes on maps like delta, the lighting doesn't even update because of the flickering because it's under a lot of stress. There was a reason why flashlights, a usually constantly moving lightsource, were converted into lighting sprites instead of using the lighting subsystem.
> * The mob spawns don't have numerical assignments, like Spider (120), and thus can cause issues when it comes to reporting players who break rules.

I'll be addressing most of these issues soon!

Gandalf2k15 commented:
> The intent of the PR is fine. Station dirt is a nice touch but it gives things for people to do, however there are some issues I think with the rest of the PR that I don't think are good for the server.
> 
> I have three issues with this PR:
> 
> * The amount of hostile mobs that spawn in maint should be limited to 1 giant spiders or bat per maint segment. I did a PR like this for Bastion of Hestia that did exactly this and it was neatly balanced and didn't turn into Security/Shaft Miners hunting mobs every round. You need to make it so that the amount of spiders and such are decently common, but not too common to the point where security cares.
> * Lighting ballast malfunctions are not a good idea in a 2D game and especially not with this lighting subsystem. I thought it was a /tg/ upstream thing at once and declared that /tg/ was probably trolling everyone with the addition, but apparently it's a Skyrat thing. The issue here is that it's very difficult to actually locate lighting ballast malfunctions as not everyone has a surround sound system. It's also very annoying to hear that constant FLANG and you don't know where it's from. From a performance perspective, it's kind of insane. as well as you're forcing the lighting engine to update constantly. Sometimes on maps like delta, the lighting doesn't even update because of the flickering because it's under a lot of stress. There was a reason why flashlights, a usually constantly moving lightsource, were converted into lighting sprites instead of using the lighting subsystem.
> * The mob spawns don't have numerical assignments, like Spider (120), and thus can cause issues when it comes to reporting players who break rules.

Lights now don't make noise, I might make them use the dynamic system.

I've updated the nest system and made it way better with cool animations and adjusted their spawn rate. So hopefully it'll be in a good position now.

BurgerLUA commented:
tbh I was willing to give this PR a chance but I don't really like it after a bunch of trial runs with it


Every round starts off with security sweeping maint, 6+ crewmembers dying because a nest was in maint and an assistant opened a door to it.

The station always looks like shit but near the 30 minute mark, an ambitious janitor manages to clean most of the station up, making the PR pointless aesthetics wise because the only thing that's left is a few rusted walls that can't reasonably be repaired.

Weird immersion breaking stuff such as a bees nest spawning next to a rats nest spawning next to a spider nest and none of them really giving a fuck until a non-simplemob comes along.

Every round I play assistant, its just me running around looking for barely alive bodies and dragging them to medical 30 minutes in.

CandleJaxx commented:
followup to previous statements, i have to agree w/ others - the station seems far more dirty than clean- places w/out many props (chem, holodeck, among others) have a 90% dirty to 10% clean tile ratio- it would make far more sense in reverse. all recent rounds also immediately start with a code blue, allowing security to bypass the tazers entirely as noone enforces weapon reliquishment

Useroth commented:
Let's go ahead with it for now cause we've got testmerge conflicts. This will be worked on and improved.
